daffodil-core/src/main/scala/org/apache/daffodil/runtime1/ChoiceTermRuntime1Mixin.scala

```diff
@@ -31,9 +31,12 @@ import org.apache.daffodil.api.WarnID
 import org.apache.daffodil.processors.ChoiceDispatchKeyEv
 import org.apache.daffodil.dsom.ExpressionCompilers
 import org.apache.daffodil.dpath.NodeInfo
+import org.apache.daffodil.grammar.Gram
 
 trait ChoiceTermRuntime1Mixin { self: ChoiceTermBase =>
 
+  protected def alternatives: Seq[Gram]
```

Comment:
Add scala doc. The alternatives are the members of the choice group with special treatment given to some kinds of members. An invariant is that if a direct child member is an array element, the child will have been encapsulated as a sequence, so that arrays always live within sequences.